ProductsManegements/Inventory.cs

```diff
@@ -0,0 +1,198 @@
+            do
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("Enter the quantity of the product: ");
+                Console.ResetColor();
+
+            } while (!int.TryParse(Console.ReadLine(), out quantity));
+
+            product=new Product(name, price, quantity);
+            products.Add(product);
+
+            Console.ForegroundColor = ConsoleColor.Green;
+            Console.WriteLine($"A new product {product.Name} is added successfully! ");
+            Console.ResetColor();
+        }
+
+        public void ViewAllProducts()
+        {
+           
+            foreach(Product product in products)
+            {
+                Console.ForegroundColor = ConsoleColor.Yellow;
+                Console.WriteLine("-----------------------");
+                Console.ResetColor();
+                Console.WriteLine(product.ShowProductDetails());
+       
+            }
+
+
+        }
+
+        public Product? SearchProduct()
+        {
+            Console.ForegroundColor= ConsoleColor.Green;
+            Console.WriteLine("Enter the product name:");
+            Console.ResetColor();
+            string ?name =Console.ReadLine();
+            return products.Find(x => x.Name == name);
+
+        }
+
+        public void SearchDisplayProduct()
+        {
+            Product? product = SearchProduct();
+         
+            if (product != null)
+            {
+                Console.WriteLine(product.ShowProductDetails());
+            }
+            else
+            {
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("No such product! Recheck the name and try again!");
+                Console.ResetColor();
+            }
+        }
+
+        public void DeleteProduct()
+        {
+            Product? product = SearchProduct();
+
+            if (product != null)
+            {
+                products.Remove(product);
+                Console.ForegroundColor = ConsoleColor.Green;
+                Console.WriteLine("Product deleted Sucessfully!");
+                Console.ResetColor();
+            }
+            else
+            {
+
+                Console.ForegroundColor = ConsoleColor.Red;
+                Console.WriteLine("No such product! Recheck the name and try again!");
+                Console.ResetColor();
+            }
+        }
+
+        public void UpdateProduct()
+        {
+            Product? product=SearchProduct();
+            //string pattern = @"^(?![YyNn]$).*$";
+            int input =0;
+            if (product != null)
+            {
+               void menu()
+                {
+                    Console.WriteLine(@"
+------------------------
+Choose what to modify ->
+------------------------
+");
+                    int i = 1;
+                    foreach (PropertyInfo p in product.GetType().GetProperties())
```

Comment:
Great that you've start using reflection before reaching it in the plan, but notice that if we didn't use it wisely it's going to increase the accidental complexity of our code.